cmd/kubectl-gadget/main.go

```diff
@@ -72,34 +77,36 @@ func main() {
 		}
 	}
 
-	runtime := grpcruntime.New(grpcruntime.WithConnectUsingK8SProxy)
-	runtimeGlobalParams := runtime.GlobalParamDescs().ToParams()
-	common.AddFlags(rootCmd, runtimeGlobalParams, nil, runtime)
-	runtime.Init(runtimeGlobalParams)
+	grpcRuntime = grpcruntime.New(grpcruntime.WithConnectUsingK8SProxy)
+	runtimeGlobalParams = grpcRuntime.GlobalParamDescs().ToParams()
+	common.AddFlags(rootCmd, runtimeGlobalParams, nil, grpcRuntime)
+	grpcRuntime.Init(runtimeGlobalParams)
+
+	// evaluate flags early for runtimeGlobalFlags; this will make
+	// sure that --connection-method has already been parsed when calling
+	// InitDeployInfo()
+
+	err := commonutils.ParseEarlyFlags(rootCmd)
+	if err != nil {
+		// Analogous to cobra error message
+		fmt.Fprintf(os.Stderr, "Error: %v\n", err)
+		os.Exit(1)
+	}
+
```

Comment:
Its needed for the following purpose:
- We call `kubectl-gadget deploy --gadget-namespace=foobar`
- Since we used `deploy`, where `skipInfo==true`, we are not parsing any flags
- In the step where we actually want to deploy the gadget pods, we now don't have the namespace information

Same for `undeploy`